delira/_backends.py

```diff
@@ -0,0 +1,73 @@
+import os
+import json
+from delira._version import __version__
+
+# to register new possible backends, they have to be added to this list.
+# each backend should consist of a tuple of length 2 with the first entry
+# being the package import name and the second being the backend abbreviation.
+# E.g. TensorFlow's package is named 'tensorflow' but if the package is found,
+# it will be considered as 'tf' later on
+__POSSIBLE_BACKENDS = [("torch", "torch"), ("tensorflow", "tf")]
+__BACKENDS = []
+
+
+def _determine_backends():
+
+    _config_file = __file__.replace("_backends.py", ".delira")
+    # look for config file to determine backend
+    # if file exists: load config into environment variables
+
+    if not os.path.isfile(_config_file):
+        _backends = {}
+        # try to import all possible backends to determine valid backends
+
+        import importlib
+        for curr_backend in __POSSIBLE_BACKENDS:
+            try:
+                assert len(curr_backend) == 2
+                assert all([isinstance(_tmp, str) for _tmp in curr_backend]), \
+                    "All entries in current backend must be strings"
+
+                # check if backend can be imported
+                bcknd = importlib.util.find_spec(curr_backend[0])
+
+                if bcknd is not None:
+                    _backends[curr_backend[1]] = True
+                else:
+                    _backends[curr_backend[1]] = False
+                del bcknd
+
+            except ValueError:
+                _backends[curr_backend[1]] = False
+
+        with open(_config_file, "w") as f:
+            json.dump({"version": __version__, "backend": _backends},
+                      f, sort_keys=True, indent=4)
+
+        del _backends
+
+    # set values from config file to variable
+    with open(_config_file) as f:
+        _config_dict = json.load(f)
+    for key, val in _config_dict.pop("backend").items():
+        if val:
+            __BACKENDS.append(key.upper())
```

Comment:
Any reason why it is `upper()` and not `lower()`? I would prefer `lower()` to be consistent with `__POSSIBLE_BACKENDS ` at the top.

Reply:
It has been upper, because this somehow are constants we are checking in the rest of the code, and by default, constants should be uppercase. What do the others think on this? @haarburger @ORippler 

Also this is the same as done before :) It's just some refactoring, no breaking changes

Reply:
So I'll merge this now with uppercase, if we want to change this, we can still do this somewhere else